demos/CompositeUI-MVC/Divergent.Frontend/App_Start/ContainerConfig.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Castle.MicroKernel.Registration;
+using Castle.MicroKernel.Resolvers.SpecializedResolvers;
+using Castle.Windsor;
+using Divergent.ITOps.ViewModelComposition;
+using System;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Web;
+using System.Web.Mvc;
+using System.Web.Optimization;
+
+namespace Divergent.Frontend
+{
+    public class ContainerConfig
+    {
+        public static IWindsorContainer ConfigContainer()
+        {
+            var container = new WindsorContainer();
+            container.Kernel.Resolver.AddSubResolver(new ArrayResolver(container.Kernel, true));
+
+            var bin = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "bin");
+            var concreteTypes = Directory.EnumerateFiles(bin, "Divergent*.dll")
+                .SelectMany(path => Assembly.LoadFile(path).GetTypes())
+                .Where(t => !t.IsInterface && !t.IsAbstract);
```

Comment:
The `concreteTypes` variable (and it dependency `bin`) are not used until the `foreach` block, it would be nicer if these two assignments could be moved below the next `Register` call, so that they sit next to the `foreach` block.